docs/content/basic/assistant.md

```diff
@@ -0,0 +1,238 @@
+        for message in replies_in_thread["messages"]:
+            role = "user" if message.get("bot_id") is None else "assistant"
+            messages_in_thread.append({"role": role, "content": message["text"]})
+
+        # Pass the latest prompt and chat history to the LLM (call_llm is your own code)
+        returned_message = call_llm(messages_in_thread)
+
+        # Post the result in the assistant thread
+        say(text=returned_message)
+
+    except Exception as e:
+        logger.exception(f"Failed to respond to an inquiry: {e}")
+        # Don't forget sending a message telling the error
+        # Without this, the status 'is typing...' won't be cleared, therefore the end-user is unable to continue the chat
+        say(f":warning: Sorry, something went wrong during processing your request (error: {e})")
+
+# Enable this assistant middleware in your Bolt app
+app.use(assistant)
+```
+
+Refer to [the module document](https://tools.slack.dev/bolt-python/api-docs/slack_bolt/kwargs_injection/args.html) to learn the available listener arguments.
+
+When an end-user opens an assistant thread next to a channel, the channel information is stored as the thread's `AssistantThreadContext` data, and you can access this information by using `get_thread_context` utility.
+
+When the user switches the channel, the `assistant_thread_context_changed` event will be sent to your app. If you use the built-in `Assistant` middleware without any custom configuration (like the above code snippet does), the updated context data is automatically saved as message metadata of the first reply from the assistant bot. This means that, as long as you go with this built-in approach, you don't need to store the context data within any datastore. The only downside of this default module is the runtime overhead of additional Slack API calls. More specifically, it calls `conversations.history` API to look up the stored message metadata when you execute `get_thread_context`.
+
+If you prefer storing this data elsewhere, you can pass your own `AssistantThreadContextStore` implementation to the `Assistant` constructor. We provide a reference implementation using the local file system, named `FileAssistantThreadContextStore`:
+
+```python
+# You can use your own thread_context_store if you want
+from slack_bolt import FileAssistantThreadContextStore
+assistant = Assistant(thread_context_store=FileAssistantThreadContextStore())
+```
+
+Since this reference implementation relies on local files, it's not recommended for production-grade operations. For your production apps, please create your own class that inherits `AssistantThreadContextStore`.
+
+<details>
+
+<summary>
+Block Kit interactions in the assistant thread
+</summary>
+
+For advanced use cases, you might want to use Block Kit buttons instead of the suggested prompts. Additionally, consider sending a message with structured metadata to trigger subsequent interactions with the user.
```

Comment:
Maybe we link to message metadata docs here?
```suggestion
For advanced use cases, you might want to use Block Kit buttons instead of the suggested prompts. Additionally, consider sending [a message with structured metadata](https://api.slack.com/metadata) to trigger subsequent interactions with the user.
```